### PR TITLE
pylint: E1507 invalid-envvar-value

### PR DESCRIPTION
**Pull request by latonis**
Rule reference: https://pycodequ.al/docs/pylint-messages/e1507-invalid-envvar-value.html

Implemented for Pylint parent issue: #970

### CODE REVIEW

File: crates/ruff/resources/test/fixtures/pylint/invalid_envvar_value.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import os
+
+os.getenv(1) # [invalid-envvar-value]
+os.getenv("a")
+os.getenv('test')
+
+os.getenv(["hello"]) # [invalid-envvar-value]
+
+AA = "aa"
+os.getenv(AA)
```
Comment (charliermarsh) [inline crates/ruff/resources/test/fixtures/pylint/invalid_envvar_value.py:10]: Nit: can you add a newline here? Or better yet, run this file through Black?

File: crates/ruff/src/rules/pylint/rules/invalid_envvar_value.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+/// PLE1507
+pub fn invalid_envvar_value(checker: &mut Checker, func: &Expr, args: &[Expr]) {
+    if checker
+        .ctx
+        .resolve_call_path(func)
+        .map_or(false, |call_path| call_path.as_slice() == ["os", "getenv"])
+    {
+        // Get the first argument for `getenv`
+        if let Some(expr) = args.get(0) {
```
Comment (charliermarsh) [inline crates/ruff/src/rules/pylint/rules/invalid_envvar_value.rs:44]: I think `os.getenv(key="foo", default_value="bar")` is also valid. Can we also check if `key` is provided as a kwarg?

**Review by charliermarsh — CHANGES_REQUESTED** [review]
Thanks!

File: crates/ruff/resources/test/fixtures/pylint/invalid_envvar_value.py (hunk 1 above)
Comment (latonis) [reply crates/ruff/resources/test/fixtures/pylint/invalid_envvar_value.py]: Fixed :)

File: crates/ruff/src/rules/pylint/rules/invalid_envvar_value.rs (hunk 2 above)
Comment (latonis) [reply crates/ruff/src/rules/pylint/rules/invalid_envvar_value.rs]: Implemented :)

**Comment by latonis** [comment]
> information_source ecosystem check **detected changes**. (+0, -360, 0 error(s))
> 
> zulip (+0, -2)
> ```diff
> - zerver/apps.py:32:9: B018 Found useless attribute access. Either assign it to a variable or remove it.
> - zerver/views/realm.py:185:5: B018 Found useless expression. Either assign it to a variable or remove it.
> ```
> 
> bokeh (+0, -309)
> airflow (+0, -49)

Changes have been implemented @charliermarsh , but I'm unsure why these changes happened in the ecosystem checks, I don't believe this has anything to do with my changes?

**Comment by charliermarsh** [comment]
@latonis - Yeah all good -- there's something wrong with the base-branch comparison.

**Pull request merged**